src/process_controller.cpp

```diff
@@ -219,6 +218,14 @@ void ProcessController::handle_ptrace_event(pid_t child, int event)
         }
     }
     break;
+    case PTRACE_EVENT_EXEC:
+    {
+        pid_t new_pid = ptrace_geteventmsg(child);
+        std::string name = get_process_comm(child);
```

Comment:
We don't need this at all, child is already the pid that we want.

Reply:
I understand the documentation differently:

> If the execing thread is not a thread group leader, the thread ID is reset to thread group leader's ID before this stop. Since Linux 3.0, the former thread ID can be retrieved with PTRACE_GETEVENTMSG. 

If I read this correctly, `child` is only correct for single-thread processes. But if a thread calls exec, this pid will be wrong.

Side note: Using PTRACE_GETEVENTMSG in this case will bump our Linux dependency subtly to 3.0.

Reply:
having exec1.cpp as:
```cpp
#include <iostream>
#include <thread>
#include <unistd.h>
#include <sys/types.h>
#include <errno.h>
#include <string.h>

void work()
{
    std::cerr << "pid: " << getpid() << std::endl;
    std::cerr << "tid: " << gettid() << std::endl;
    const char *name = "exec2";
    execlp(name, name, NULL);
    std::cerr << strerror(errno) << std::endl;
}
int main (int argc, char **argv)
{
    std::thread t1(work);
    t1.join();
    return 0;
}
```
and exec2.cpp as
```cpp
#include <iostream>
#include <sys/types.h>
#include <unistd.h>

int main(int argc, char ** argv)
{
    std::cerr << "exec pid: " << getpid() << std::endl;
    std::cerr << "exec tid: " << gettid() << std::endl;
    return 0;
}
```
Returns something like:

```
pid: 3161
tid: 3162
exec pid: 3161
exex tid: 3161
```

So, as far as I understand it, we don't need the former tid,only the tgid (pid afaik)

Reply:
Thanks for the code. I see, the important part, which I forogot is that the whole process gets replaced, so the main thread seizes to exists.